xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -4709,7 +4695,7 @@ public void setAttachmentList(List<XWikiAttachment> list)
         }
     }
 
-    public List<XWikiAttachment> getAttachmentList()
+    public XWikiAttachmentList getAttachmentList()
```

Comment:
This actually breaks an API. XWikiAttachmentList must remain internal, what is important for outside world is that it impléments List<XWikiAttachment>. By the way I'm pretty sure this change produce a build error.